src/app-router.ts

```diff
@@ -96,10 +97,13 @@ export class AppRouter extends Router {
           const resolveConfiguredPromise = this._resolveConfiguredPromise;
           this._resolveConfiguredPromise = () => {/**/};
           return this
-            .configure(config => {
-              viewModel.configureRouter(config, this);
-              return config;
-            })
+            .configure(config =>
+              Promise
+                .resolve(viewModel.configureRouter(config, this))
+                // an issue with configure interface. Should be fixed there
+                // todo: fix this via configure interface in router
+                .then(() => config) as any
```

Comment:
If cast to any, why not just
```js
.configure(config => viewModel.configureRouter(config, this) as any)
```

Reply:
It still expects a `RouterConfiguration` or a `Promise<RouterConfiguration>`, and what is returned there is a `Promise<RouterConfiguration>`

I think returning the right thing is still better than relying on the return of `configureRouter` of viewModel